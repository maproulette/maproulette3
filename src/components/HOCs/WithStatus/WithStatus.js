import { connect } from 'react-redux'
import _get from 'lodash/get'
import { FETCHING_CHALLENGES_STATUS,
         CHECKING_LOGIN_STATUS }
       from '../../../services/Status/Status'

const mapStateToProps = state => {
  return {
    fetchingChallenges:
      _get(state, `currentStatus.${FETCHING_CHALLENGES_STATUS}`, []),

    checkingLoginStatus:
      _get(state, `currentStatus.${CHECKING_LOGIN_STATUS}`, false),
  }
}

export default WrappedComponent =>
  connect(mapStateToProps)(WrappedComponent)
