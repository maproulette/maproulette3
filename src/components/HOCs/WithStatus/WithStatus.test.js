import { Component } from 'react'
import { mapStateToProps, mapDispatchToProps } from './WithStatus'
import { FETCHING_CHALLENGES_STATUS,
         CHECKING_LOGIN_STATUS }
       from '../../../services/Status/Status'

let basicState = null

beforeEach(() => {
  basicState = {
    currentStatus: {}
  }
})

test("mapStateToProps maps fetchingChallenges", () => {
  const fetchIds = ["fetch123", "fetch456"]
  basicState.currentStatus[FETCHING_CHALLENGES_STATUS] = fetchIds

  const mappedProps = mapStateToProps(basicState)
  expect(mappedProps.fetchingChallenges).toEqual(fetchIds)

  expect(mappedProps).toMatchSnapshot()
})

test("fetchingChallenges returns empty array if no status", () => {
  const mappedProps = mapStateToProps(basicState)
  expect(mappedProps.fetchingChallenges).toEqual([])

  expect(mappedProps).toMatchSnapshot()
})

test("mapStateToProps maps checkingLoginStatus", () => {
  basicState.currentStatus[CHECKING_LOGIN_STATUS] = true

  const mappedProps = mapStateToProps(basicState)
  expect(mappedProps.checkingLoginStatus).toEqual(true)

  expect(mappedProps).toMatchSnapshot()
})

test("checkingLoginStatus returns false if no status", () => {
  const mappedProps = mapStateToProps(basicState)
  expect(mappedProps.checkingLoginStatus).toEqual(false)

  expect(mappedProps).toMatchSnapshot()
})
