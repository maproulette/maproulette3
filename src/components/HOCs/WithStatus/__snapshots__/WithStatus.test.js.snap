// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`checkingLoginStatus returns false if no status 1`] = `
{
  "checkingLoginStatus": false,
  "fetchingChallenges": [],
}
`;

exports[`fetchingChallenges returns empty array if no status 1`] = `
{
  "checkingLoginStatus": false,
  "fetchingChallenges": [],
}
`;

exports[`mapStateToProps maps checkingLoginStatus 1`] = `
{
  "checkingLoginStatus": true,
  "fetchingChallenges": [],
}
`;

exports[`mapStateToProps maps fetchingChallenges 1`] = `
{
  "checkingLoginStatus": false,
  "fetchingChallenges": [
    "fetch123",
    "fetch456",
  ],
}
`;
